pkg/api/vm/handler.go

```diff
@@ -397,6 +429,85 @@ func (h *vmActionHandler) abortMigration(namespace, name string) error {
 	return nil
 }
 
+func (h *vmActionHandler) findMigratableNodes(rw http.ResponseWriter, namespace, name string) error {
+	vmi, err := h.vmiCache.Get(namespace, name)
+	if err != nil {
+		return err
+	}
+	nodes, err := h.findMigratableNodesByVMI(vmi)
+	if err != nil {
+		return err
+	}
+	resp := GetMigratableNodesOutput{
+		Nodes: nodes,
+	}
+
+	util.ResponseOKWithBody(rw, resp)
+	return nil
+}
+
+func (h *vmActionHandler) findMigratableNodesByVMI(vmi *kubevirtv1.VirtualMachineInstance) ([]string, error) {
+	nodeSelector := labels.Everything()
+
+	// get node selector from network
+	vmNetworks := vmi.Spec.Networks
+	for _, network := range vmNetworks {
+		requirement, err := h.getNodeSelectorRequirementFromNetwork(vmi.Namespace, network)
+		if err != nil {
+			return nil, err
+		}
+		if requirement != nil {
+			nodeSelector = nodeSelector.Add(*requirement)
+		}
+	}
+	nodes, err := h.nodeCache.List(nodeSelector)
+	if err != nil {
+		return nil, err
+	}
+	if len(nodes) == 0 {
+		return nil, nil
+	}
+
+	// ignore the node where the VM is running
+	var migratableNodes = make([]string, 0, len(nodes)-1)
+	for _, node := range nodes {
+		if vmi.Status.NodeName == node.Name {
+			continue
+		}
+		migratableNodes = append(migratableNodes, node.Name)
+	}
+	return migratableNodes, nil
+}
+
+func (h *vmActionHandler) getNodeSelectorRequirementFromNetwork(defaultNamespace string, net kubevirtv1.Network) (*labels.Requirement, error) {
```

Comment:
You should be able to use the existing rules from the VM/VMI; you also need to add the user's custom selector rules into account.